src/gui/sdlmain.cpp

```diff
@@ -1763,10 +1764,15 @@ static void GUI_ShutDown(Section *)
 
 static void SetPriority(PRIORITY_LEVELS level)
 {
-	// TODO replace platform-specific API with SDL_SetThreadPriority
+	// Just let the OS scheduler manage priority
+	if (level == PRIORITY_LEVEL_AUTO)
+		return;
+
+		// TODO replace platform-specific API with SDL_SetThreadPriority
 #if defined(HAVE_SETPRIORITY)
-// Do nothing if priorties are not the same and not root, else the highest
-// priority can not be set as users can only lower priority (not restore it)
+	// If the priorities are different, do nothing unless the user is root,
+	// since priority can always be lowered but requires elevated rights 
+	// to increase
```

Comment:
Huge improvement. I can now casually read this and understand it, first-pass.
Before, I had to back-track and think through the logic (several times, infact).